### ISSUE TITLE
feat: [M-Perf-E1] True P/D split — include logits in state blob to eliminate 1-token trick

### ISSUE THREAD

**Issue by ddvnguyen** [ddvnguyen/hydra_vortex#266]
## Problem

The current `EnginePrefill (0x42)` / `StatePut (0x31)` / `EngineDecode (0x43)` cycle still has an implicit dependency on the 1-token trick from the old llama-server design.

`llama_state_seq_get_data` serialises the KV cache (K, V tensors per layer) but **not the output logits tensor**. After `StatePut` restores the KV cache on the decode GPU, `ctx->logits` is uninitialised — calling `common_sampler_sample` immediately reads garbage and the first generated token is wrong.

The DECODE cross-GPU handler calls `common_sampler_sample` as the very first operation:

```cpp
while (n_decoded < n_predict) {
    llama_token id = common_sampler_sample(slot->smpl.get(), ctx_tgt, -1);
    // ↑ logits are garbage after StatePut — first token is undefined behaviour
```

## Root cause

P/D contract is incomplete. PREFILL computes logits for the last prompt token (`need_logits=true`) but they live in `ctx->logits` on the prefill GPU and are not included in the state blob. The decode GPU never sees them.

## Fix — extend the state blob with the last-position logits

**Wire format change (additive, backward-compatible):**

PREFILL response:
- Meta: `{"n_past": N, "state_size": N, "logits_size": M}`
- Payload: KV blob (N bytes) + logits (M bytes = `n_vocab * sizeof(float)`)

StatePut (0x31) — extend handler:
- After `llama_state_seq_set_data` consumes KV bytes, if payload has remaining bytes equal to `n_vocab * sizeof(float)`, copy them into `ctx->logits` directly
- Old StatePut payloads (KV only) continue to work unchanged

DECODE cross-GPU:
- No change needed — `common_sampler_sample` works immediately because `ctx->logits` is valid after StatePut

**Why this is the correct boundary:**

The logits for the last prompt token are the "activation handoff" in P/D disaggregation literature (Splitwise, DistServe). They are the output of the prefill phase and the input to the decode phase. Including them in the state blob makes the contract explicit:

- Prefill contract: prompt → (KV state, logits for first decode step)
- Decode contract: (KV state + logits) → token stream

**Size:** `n_vocab * 4` bytes ≈ 600 KB for Qwen3.5 MoE (152K vocab) — negligible next to 800 MB KV.

## Files to change (fork only — no C# protocol changes needed)

- `tools/server/server-context.cpp` — PREFILL: extract logits after `llama_decode`; StatePut: inject logits into `ctx->logits` from trailing bytes
- `tools/server/server-rpc.cpp` — PREFILL RPC response: append logits bytes after KV blob; update meta to include `logits_size`

The KV+logits blob is opaque to Hydra.Core and the Store — it flows through unchanged.

Part of #161 (E1 engine — true P/D split).
